sys/include/net/ieee802154/radio.h

```diff
@@ -0,0 +1,1085 @@
+     *
+     * @return true if the radio supports the cap
+     * @return false otherwise
+     */
+    bool (*get_cap)(ieee802154_dev_t *dev, ieee802154_rf_caps_t cap);
+
+    /**
+     * @brief Set the threshold for the Energy Detection (first mode of CCA)
+     *
+     * @pre the device is on
+     *
+     * @param[in] dev IEEE802.15.4 device descriptor
+     * @param[in] threshold the threshold in dBm.
+     *
+     * @return 0 on success
+     * @return negative errno on error
+     */
+    int (*set_cca_threshold)(ieee802154_dev_t *dev, int8_t threshold);
+
+    /**
+     * @brief Set CCA mode
+     *
+     * All radios MUST at least implement the first CCA mode (ED Threshold).
+     *
+     * @pre the device is on
+     *
+     * @param[in] dev IEEE802.15.4 device descriptor
+     * @param[in] mode the CCA mode
+     *
+     * @return 0 on success
+     * @return -ENOTSUP if the mode is not supported
+     * @return negative errno on error
+     */
+    int (*set_cca_mode)(ieee802154_dev_t *dev, ieee802154_cca_mode_t mode);
+
+    /**
+     * @brief Set IEEE802.15.4 PHY configuration (channel, TX power)
+     *
+     * This function SHOULD NOT validate the PHY configurations unless
+     * it's specific to the device. The upper layer is responsible of all kind
+     * of validations.
+     *
+     * @pre the device is on
+     * @post the upper layer calls @ref
+     * ieee802154_radio_ops::request_set_trx_state immediately after calling
+     * this function.
+     *
+     * @param[in] dev IEEE802.15.4 device descriptor
+     * @param[in] conf the PHY configuration
+     *
+     * @return 0 on success
+     * @return negative errno on error
+     */
+    int (*config_phy)(ieee802154_dev_t *dev, ieee802154_phy_conf_t *conf);
+
+    /**
+     * @brief Set IEEE802.15.4 promiscuous mode
+     *
+     * @pre the device is on
+     *
+     * @note this function pointer can be NULL if the device doesn't support
+     *       hardware address filtering.
+     *
+     * @param[in] dev IEEE802.15.4 device descriptor
+     * @param[in] enable whether the promiscuous mode should be enabled or not.
+     *
+     * @return 0 on success
+     * @return negative errno on error
+     */
+    int (*set_promiscuous)(ieee802154_dev_t *dev, bool enable);
+
+    /**
+     * @brief Set IEEE802.15.4 addresses in hardware address filter
+     *
+     * @pre the device is on
+     *
+     * @note this function pointer can be NULL if the device doesn't support
+     *       hardware address filtering.
+     *
+     * @param[in] dev IEEE802.15.4 device descriptor
+     * @param[in] short_addr the IEEE802.15.4 short address
+     * @param[in] ext_addr the IEEE802.15.4 extended address
+     * @param[in] pan_id the IEEE802.15.4 PAN ID
+     *
+     * @return 0 on success
+     * @return negative errno on error
+     */
+    int (*set_hw_addr_filter)(ieee802154_dev_t *dev, uint8_t *short_addr,
+                              uint8_t *ext_addr, uint16_t pan_id);
```

Comment:
No `eui64_t`?

Reply:
I will update it

Reply:
https://github.com/RIOT-OS/RIOT/pull/14371/commits/9e7a82e704e8847e8a2ea5b6af13c7c3f47070eb